kolibri/plugins/coach/api.py

```diff
@@ -192,14 +192,18 @@ def get_queryset(self):
         notifications_query = self.apply_group_filter(notifications_query)
         after = self.check_after()
```

Comment:
It feels like it makes sense to me for the `after` value to no longer be an id, but rather be a timestamp, that way we can avoid the indirection you've had to deal with below.

Reply:
Oh I did it to keep the same API :sweat_smile:. And also didnt found any example of how do we send timestamps as filters to the rest api. Do we send a unix timestamp from the frontend? Or the same ISO date format we send from the backend? And after that use something like `after = datetime.datetime.fromisoformat(notifications_after)`? If there is any example of other place where we do something similar woul appreciate it :open_hands:.

Reply:
You should just be able to pass a Javascript Date object as a parameter to the request, and it should get properly serialized.

We may want to update the filtering that is happening here to use a django_filter Filterset (similarly to what we do in other places) as this will handle properly turning the appropriate filter values into the right sort of objects.

For filtering on a datetime, we would use the DateTimeFilter: https://django-filter.readthedocs.io/en/stable/ref/filters.html#datetimefilter

Unfortunately, there aren't any examples of using this specific filter in the codebase, but for example you would use a field definition like this:

```
before = DateTimeFilter(field_name="timestamp", lookup_expr="lt")
after = DateTimeFilter(field_name="timestamp", lookup_expr="gt")
```

For a fuller example of using a FilterSet, see here: https://github.com/learningequality/kolibri/blob/2692958f7772b099cf9191ff47e685ff4ca83206/kolibri/core/content/api.py#L421

(you shouldn't have to use methods for most of your filtering)